src/mcts/search.cc

```diff
@@ -710,6 +715,21 @@ void SearchWorker::GatherMinibatch() {
   int minibatch_size = 0;
   int collision_events_left = params_.GetMaxCollisionEvents();
   int collisions_left = params_.GetMaxCollisionVisitsId();
+  if (params_.GetMaxConcurrentSearchers() != 0) {
+    while (true) {
+      if (search_->stop_.load(std::memory_order_acquire)) return;
+      int available =
+          search_->pending_searchers_.load(std::memory_order_acquire);
+      if (available > 0 &&
+          search_->pending_searchers_.compare_exchange_weak(
+              available, available - 1, std::memory_order_acq_rel)) {
+        break;
+      }
+      // This is a hard spin lock to reduce latency but at the expense of busy
```

Comment:
Alternatively, if values more than 1 are not useful, it can be just a mutex lock. :-P

And I kind of doubt that busy wait is faster that cv or mutex.

Reply:
I could see reason to think values up to the maximum available backend concurrency are plausible, although 1 certainly seems the most promising value.
Busy wait is easily as fast or faster than cv or mutex in best case, its not possible to reliably pass information between cpu cores faster than via atomic checks on memory, and under the assumption the machine has cores to spare, the thread doesn't have to yield with risk of having to wait for OS to reschedule it onto a core with all the associated register backup and restore.
Busy wait may on the other hand not always be the best choice (especially in scenarios other than multiple equivalent waiters, since spinlock isn't a fair lock).  But I think under the cases where this actually potentially adds value, busy wait downsides aren't likely to come into play.